cpu/samd5x/include/candev_samd5x.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2024 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup         cpu_samd5x
+ * @brief           CPU specific definitions for CAN controllers
+ * @{
+ *
+ * @file
+ * @brief           SAMD5x CAN controller (M_CAN Bosch) driver
+ *
+ * @author          Firas Hamdi <firas.hamdi@ml-pa.com>
+ */
+
+#ifndef CANDEV_SAMD5X_H
+#define CANDEV_SAMD5X_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(CAN_INST_NUM)
+#include "can/candev.h"
+
+#ifndef CANDEV_SAMD5X_DEFAULT_BITRATE
+/** Default bitrate */
+#define CANDEV_SAMD5X_DEFAULT_BITRATE 500000U
+#endif
+
+#ifndef CANDEV_SAMD5X_DEFAULT_SPT
+/** Default sampling-point */
+#define CANDEV_SAMD5X_DEFAULT_SPT 875
+#endif
+
+#ifndef CANDEV_SAMD5X_DEFAULT_STD_FILTER_NUM
+#define CANDEV_SAMD5X_DEFAULT_STD_FILTER_NUM 3
+#endif
+
+#ifndef CANDEV_SAMD5X_DEFAULT_EXT_FILTER_NUM
+#define CANDEV_SAMD5X_DEFAULT_EXT_FILTER_NUM 3
+#endif
+
+#ifndef CANDEV_SAMD5X_DEFAULT_RX_FIFO_0_ELTS_NUM
+#define CANDEV_SAMD5X_DEFAULT_RX_FIFO_0_ELTS_NUM 32
+#endif
+
+#ifndef CANDEV_SAMD5X_DEFAULT_RX_FIFO_1_ELTS_NUM
+#define CANDEV_SAMD5X_DEFAULT_RX_FIFO_1_ELTS_NUM 32
+#endif
+
+#ifndef CANDEV_SAMD5X_DEFAULT_TX_EVT_FIFO_ELTS_NUM
+#define CANDEV_SAMD5X_DEFAULT_TX_EVT_FIFO_ELTS_NUM 16
+#endif
+
+#ifndef CANDEV_SAMD5X_DEFAULT_TX_BUFFER_NUM
+#define CANDEV_SAMD5X_DEFAULT_TX_BUFFER_NUM 16
+#endif
+
+#ifndef CANDEV_SAMD5X_DEFAULT_TX_BUFFER_FIFO_QUEUE_NUM
+#define CANDEV_SAMD5X_DEFAULT_TX_BUFFER_FIFO_QUEUE_NUM 16
+#endif
+
+/* unit: elements */
+#define CANDEV_SAMD5X_MAX_STD_FILTER 128
+#define CANDEV_SAMD5X_MAX_EXT_FILTER 64
+#define CANDEV_SAMD5X_MAX_RX_FIFO_0_ELTS 64
+#define CANDEV_SAMD5X_MAX_RX_FIFO_1_ELTS 64
+#define CANDEV_SAMD5X_MAX_RX_BUFFER 64
+#define CANDEV_SAMD5X_MAX_TX_EVT_FIFO_ELTS 32
+#define CANDEV_SAMD5X_MAX_TX_BUFFER 32
+#define CANDEV_SAMD5X_MSG_RAM_MAX_SIZE 448
+
+/**
+ * @brief CAN device configuration descriptor
+ */
+typedef struct {
+    Can *can;                                       /**< CAN device handler */
+    gpio_t rx_pin;                                  /**< CAN Rx pin */
+    gpio_t tx_pin;                                  /**< CAN Tx pin */
+    uint8_t gclk_src;                               /**< GCLK source supplying the CAN controller */
```

Comment:
```suggestion
    Can *can;           /**< CAN device handler */
    gpio_t rx_pin;      /**< CAN Rx pin */
    gpio_t tx_pin;      /**< CAN Tx pin */
    uint8_t gclk_src;   /**< GCLK source supplying the CAN controller */
```

Reply:
there is no reason to halve comments touching the right side of the editor if there isnt even a thing they align with

Reply:
Moved the comments above the members